### ISSUE TITLE
Refactor common testing code

### ISSUE THREAD

**Issue by Horneth** [broadinstitute/cromwell#1581]
When switching to the last version of WDL4s, one noticeable change is that a task cannot share its name with the workflow name. I realized that there are close to a dozen "hello world" WDLs in the test suite scattered across projects but also duplicated within the same project. This probable is somehow close to the idea of having a canonical test suite for backends that we talked about before.

**Comment by geoffjentry** [comment]
That's probably going to bork a lot more stuff than just our tests.

I can see why that change was put in but I don't think we can just blindly accept that behavior w/o running it past @kcibul - it's likely to cause a lot of commotion downstream.

**Comment by Horneth** [comment]
I agree we should talk about it - although in order to enable sub-workflows, I personally think it's a good thing as it will avoid confusion (both on user and cromwell side) on what is being called.

**Comment by geoffjentry** [comment]
At the moment (he says, having only thought about it for the couple combined minutes he's had this tab open) I disagree. As long as there is no ambiguity in the fully qualified names it should be fine IMO.

**Comment by katevoss** [comment]
@Horneth is this fixed already?

**Comment by Horneth** [comment]
Nothing is broken per se. It's just good old tech debt where we have duplicated WDLs in the unit tests that could be unified. It is still there AFAIK.

**Comment by katevoss** [comment]
@Horneth does it make the unit tests a mess? Or is it just ugliness that makes us cringe slightly?

**Comment by Horneth** [comment]
Probably more the latter even though it does make the test code a bit messier.

**Comment by katevoss** [comment]
As a **Cromwell developer**, I want **each unit test to appear once**, so that I can **avoid duplicate (and messy) tests**.
- Effort: **TBD**
- Risk: **Small**
- Business value: **Small**
